### ISSUE TITLE
Callable support?

### ISSUE THREAD

**Issue by aaronjensen** [tomdalling/resonad#2]
Hi there, I've used resonad before on a project and I'm considering using it again. 

We've been making heavy use of callables lately and I was wondering if you'd ever considered enhancing resonad to work with them directly, e.g.:

```rb
update_widget = UpdateWidget.new # update_widget has a #call method

find_widget(widget_id)
  .and_then(update_widget) # or UpdateWidget, assuming UpdateWidget.call exists
```

**Comment by tomdalling** [comment]
I hadn't considered it before, but I like it. I'm all for adding conveniences.

I'll probably take a look at this on the weekend. It should be relatively easy to implement.

**Comment by tomdalling** [comment]
Implemented in v1.4 (commit 993c206e8ab5b09cfc4acf18815ad5580119bead). Let me know if I missed anything!

**Comment by aaronjensen** [comment]
Great, thank you!
